src/submission/schema/schema-manager.ts

```diff
@@ -190,6 +191,36 @@ class SchemaManager {
   };
 }
 
+export const revalidateAllDonorClinicalEntitiesAgainstSchema = (
+  donor: DeepReadonly<Donor>,
+  schema: SchemasDictionary,
+) => {
+  const clinicalSchemaNames = getSchemaNamesForDonorClinicalEntities(donor);
+  let isValid = true;
+  clinicalSchemaNames.forEach((p: any) => {
+    if (!isValid) {
+      return;
+    }
+    const errs = MigrationManager.validateDonorEntityAgainstNewSchema(p.schemaName, schema, donor);
```

Comment:
isn't `p` a `string`??

Reply:
because it comes from `clinicalSchemaNames`??

Reply:
yeah don't know how the migration test worked will fix

Reply:
this line was never called.. it was always marked as valid because of a bug in the way I iterated the enum

Reply:
the test case I added does cover this now with a negative flow